backend/adapters/secondary/gateways/client/InMemoryClientGateway.ts

```diff
@@ -19,4 +19,11 @@ export class InMemoryClientGateway implements ClientGateway {
       throw new ClientError(`No client with this id (${id})`, 404)
     return this.clients.splice(clientIndex, 1)[0]
   }
+
+  getById(id: string): Promise<Client> {
+    if (this.clients.find(x => x.id === id))
```

Comment:
You can store the result of this condition and return it instead of doing a find operation again.

Reply:
Also `x` is not a super friendly name for a client.